### PR TITLE
Handle rank-deficient LinearModel.  Also changes in docs

### PR DESCRIPTION
**Pull request by dmbates**
For reasons that I am having difficulty remembering, I incorporated documentation changes in these changes to handle rank-deficient linear models as discussed in #200.

I realize this makes it harder to review these changes for rank-deficiency.  I can start a new branch and cherry-pick the rank-deficiency-related commits if that would be preferred.

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -29,6 +30,25 @@ form = DataFrame(Any[[0.1,0.3,0.5,0.6,0.7,0.9],[0.086,0.269,0.446,0.538,0.626,0.
...
     @test isapprox(aicc(lm1), -24.409684288095946)
     @test isapprox(bic(lm1), -37.03440588041178)
+    lm2 = fit(LinearModel, hcat(ones(6), 10form[:Carb]), form[:OptDen], true)
+    @test isapprox(coef(lm1), coef(lm2) .* [1., 10.])
+end
+
+@testset "rankdeficient" begin
+    # an example of rank deficiency caused by a missing cell in a table
+    dfrm = DataFrame([categorical(repeat(string.('A':'D'), inner = 6)),
+               categorical(repeat(string.('a':'c'), inner = 2, outer = 4))],
```
Comment (nalimilan) [inline test/runtests.jl]: Incorrect alignment.

File: test/runtests.jl [hunk 2]
```diff
@@ -29,6 +30,25 @@ form = DataFrame(Any[[0.1,0.3,0.5,0.6,0.7,0.9],[0.086,0.269,0.446,0.538,0.626,0.
...
+    X = ModelMatrix(ModelFrame(Formula(nothing, :(1+G*H)), dfrm)).m
+    y = X * (1:size(X, 2)) + 0.1 * randn(MersenneTwister(1234321), size(X, 1))
+    inds = deleteat!(collect(1:length(y)), 7:8)
+    m1 = fit(LinearModel, X, y)
+    @test isapprox(deviance(m1), 0.28856700971719657)
+    Xmissingcell = X[inds, :]
+    ymissingcell = y[inds]
+    @test_throws LinAlg.PosDefException m2 = fit(LinearModel, Xmissingcell, ymissingcell)
+    m2p = fit(LinearModel, Xmissingcell, ymissingcell, true)
+    @test isapprox(deviance(m2p), 0.2859221258731563)
```
Comment (nalimilan) [inline test/runtests.jl]: Wouldn't it make sense to test the values of coefficients?

File: src/GLM.jl [hunk 3]
```diff
@@ -76,8 +76,18 @@ module GLM
     const FP = AbstractFloat
     const FPVector{T<:FP} = AbstractArray{T,1}
 
+"""
```
[the remark this answers is not in the document]
Comment (dmbates) [reply src/GLM.jl]: What is the correct indentation in this case?  I thought that the first line within the multi-line string should be indented 4 spaces (b/c Markdown) and the line beginning with "Abstract" should not be indented.  If I indent the triple double-quote characters things will look weird.

Perhaps the best solution is to move those abstract type declarations to linpred.jl and lm.jl.

[the remark this answers is not in the document]
Comment (nalimilan) [reply src/GLM.jl]: No, you can use whatever indentation you want (i.e. the same as surrounding lines), what matters is the alignment with the `"""`.

File: docs/make.jl [hunk 4]
```diff
@@ -1,3 +1,15 @@
-using Documenter, GLM
+using Distributions, Documenter, GLM
 
-makedocs()
+makedocs(
+    format = :html,
```
Comment (nalimilan) [inline docs/make.jl]: I think you can remove `docs/mkdocs.yml` with that option. And of course the old `docs/` folder.

Comment (dmbates) [reply docs/make.jl]: Done, thanks.

**Pull request merged**